plugin/pkg/admission/runtimeclass/admission.go

```diff
@@ -171,13 +181,43 @@ func setOverhead(a admission.Attributes, pod *api.Pod, runtimeClass *v1beta1.Run
 	return nil
 }
 
-func validateOverhead(a admission.Attributes, pod *api.Pod, runtimeClass *v1beta1.RuntimeClass) (err error) {
+func setScheduling(a admission.Attributes, pod *api.Pod, runtimeClass *v1beta1.RuntimeClass) (err error) {
+	if runtimeClass == nil || runtimeClass.Scheduling == nil {
+		return nil
+	}
+
+	// convert to internal type and assign to pod's Scheduling
+	nodeScheduling := &node.Scheduling{}
+	if err = nodev1beta1.Convert_v1beta1_Scheduling_To_node_Scheduling(runtimeClass.Scheduling, nodeScheduling, nil); err != nil {
+		return err
+	}
 
+	runtimeNodeSelector := nodeScheduling.NodeSelector
+	newNodeSelector := pod.Spec.NodeSelector
+	if newNodeSelector == nil {
+		newNodeSelector = runtimeNodeSelector
+	} else {
+		for key, value := range runtimeNodeSelector {
+			if podValue, ok := newNodeSelector[key]; ok && podValue != value {
+				return admission.NewForbidden(a, fmt.Errorf("conflict: runtimeClass.scheduling.nodeSelector[%s] = %s; pod.spec.nodeSelector[%s] = %s", key, value, key, podValue))
+			}
+			newNodeSelector[key] = value
+		}
+	}
+
+	newTolerations := tolerations.MergeTolerations(pod.Spec.Tolerations, nodeScheduling.Tolerations)
```

Comment:
@tallclair - for conflicts I'd prefer us rejecting the pod

Reply:
I actually don't think there are conflicts. If 2 tolerations "conflict", then both should be included. The logic for tolerations is a union of all tolerations. I have a fix to MergeTolerations for this (I'll send it out tomorrow).